src/sdk/strategy/cbd-strategy.ts

```diff
@@ -0,0 +1,132 @@
+import { ethers } from 'ethers';
+
+import { CbdTDecDecrypter } from '../../characters/cbd-universal-bob';
+import { Enrico } from '../../characters/enrico';
+import { ConditionSet } from '../../conditions';
+import { DkgClient, DkgRitual, DkgRitualJSON } from '../../dkg';
+import { base64ToU8Receiver, u8ToBase64Replacer } from '../../utils';
+import { Cohort, CohortJSON } from '../cohort';
+
+type StrategyJSON = {
+  cohort: CohortJSON;
+  conditionSet?: ConditionSet;
+};
+
+type DeployedStrategyJSON = {
+  dkgRitual: DkgRitualJSON;
+  cohortConfig: CohortJSON;
+  conditionSet?: ConditionSet;
+};
+
+export class CbdStrategy {
+  private constructor(
+    public readonly cohort: Cohort,
+    private readonly conditionSet?: ConditionSet
+  ) {}
+
+  public static create(cohort: Cohort, conditionSet?: ConditionSet) {
+    return new CbdStrategy(cohort, conditionSet);
+  }
+
+  public async deploy(
+    provider: ethers.providers.Web3Provider
+  ): Promise<DeployedCbdStrategy> {
+    const dkgRitualParams = {
+      threshold: this.cohort.configuration.threshold,
+      shares: this.cohort.configuration.shares,
+    };
+    const dkgClient = new DkgClient(provider);
+    const dkgRitual = await dkgClient.initializeRitual(
+      provider,
+      dkgRitualParams
+    );
+
+    const encrypter = new Enrico(
+      dkgRitual.dkgPublicKey,
+      undefined,
+      this.conditionSet
+    );
+
+    const decrypter = new CbdTDecDecrypter(
+      this.cohort.configuration.porterUri,
+      dkgRitual.dkgPublicKey
```

Comment:
Why does the decrypter need the `dkgPublicKey`?

Reply:
It doesn't, that's a refactoring artifact. I'm going to remove it.